src/dynsight/_internal/trajectory/trajectory.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+import pickle
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import MDAnalysis
+    import numpy as np
+    from numpy.typing import NDArray
+    from tropea_clustering._internal.first_classes import StateMulti, StateUni
+
+import dynsight
+
+UNIVAR_DIM = 2
+
+
+@dataclass
+class Insight:
+    """Contains an analysis perfomed on a many-body trajectory.
+
+    Attributes:
+        dataset: np.ndarray of shape (n_particles, n_frames, n_dims)
+            The values of a single-particle descriptor.
+        r_cut: float
+            The spatial cutoff with the descriptor has been computed.
+    """
+
+    dataset: NDArray[np.float64]
+    r_cut: float
```

Comment:
Remove this? Unless it is general to all "insights"

Reply:
Maybe we have another dataclass called "Meta" that contains calculation information? Then all Insights have those?